integration_tests/integration/dashboards.spec.js

```diff
@@ -410,193 +410,237 @@ describe('Dashboards', () => {
             {
-              'Date received': '13 Dec 2020',
-              Referral: 'ABCABCA2',
-              'Service user': 'Jenny Jones',
-              'Intervention type': "Women's Services - West Midlands",
-              Caseworker: 'A.Caseworker',
-              Action: 'View',
+              header: 'Referral',
+              sortedTable: [rowForReferralAWithNoCaseworkerColumn, rowForReferralBWithNoCaseworkerColumn],
+            },
+            {
+              header: 'Service user',
+              sortedTable: [rowForReferralAWithNoCaseworkerColumn, rowForReferralBWithNoCaseworkerColumn],
+            },
+            {
+              header: 'Intervention type',
+              sortedTable: [rowForReferralAWithNoCaseworkerColumn, rowForReferralBWithNoCaseworkerColumn],
             },
           ],
         },
         {
-          header: 'Caseworker',
-          sortedTable: [
+          dashboardType: 'Completed cases',
+          referrals: [completedA, completedB],
+          initialTable: [rowForReferralAWithCaseworker, rowForReferralBWithCaseworker],
+          sortedTables: [
             {
-              'Date received': '26 Jan 2021',
-              Referral: 'ABCABCA1',
-              'Service user': 'George Michael',
-              'Intervention type': 'Accommodation Services - West Midlands',
-              Caseworker: '',
-              Action: 'View',
+              header: 'Date received',
+              sortedTable: [rowForReferralAWithCaseworker, rowForReferralBWithCaseworker],
             },
             {
-              'Date received': '13 Dec 2020',
-              Referral: 'ABCABCA2',
-              'Service user': 'Jenny Jones',
-              'Intervention type': "Women's Services - West Midlands",
-              Caseworker: 'A.Caseworker',
-              Action: 'View',
+              header: 'Referral',
+              sortedTable: [rowForReferralAWithCaseworker, rowForReferralBWithCaseworker],
+            },
+            {
+              header: 'Service user',
+              sortedTable: [rowForReferralAWithCaseworker, rowForReferralBWithCaseworker],
+            },
+            {
+              header: 'Intervention type',
+              sortedTable: [rowForReferralAWithCaseworker, rowForReferralBWithCaseworker],
             },
           ],
         },
       ]
 
       const referralToSelect = sentReferralFactory.build({
         sentAt: '2021-01-26T13:00:00.000000Z',
-        referenceNumber: 'ABCABCA1',
+        referenceNumber: 'B',
         referral: {
           interventionId: accommodationIntervention.id,
           serviceUser: { firstName: 'George', lastName: 'Michael' },
         },
       })
 
-      headersAndTables.forEach(({ header, sortedTable }) => {
-        describe(`sorting by "${header}"`, () => {
-          it(`allows the user to sort by "${header}"`, () => {
-            cy.login()
+      dashBoardTables.forEach(({ dashboardType, referrals, initialTable, sortedTables }) => {
```

Comment:
How long do these tests take now, out of interest? I think this file took something like 18 seconds to run before. I guess this one of the drawbacks of E2E testing being slow. I think it's probably fine as we use these in place of unit tests for user-facing Javascript.